src/cpp/session/modules/SessionThemes.R

```diff
@@ -1027,47 +1014,114 @@
          globalPath <- normalizePath(.rs.getThemeInstallDir(TRUE), mustWork = FALSE, winslash = "/")
          if (identical(file.path(globalPath, justFileName), actualPath))
          {
-            stop(
-               "Unable to remove the specified theme \"",
-               name,
-               "\", which is installed for all users. Please contact your system administrator.")
+            stop("The specified theme is installed for all users. Please contact your system administrator to remove the theme.")
          }
          else
          {
-            stop(
-               "Unable to remove the specified theme \"",
-               name,
-               "\". Please check your file system permissions.")
+            stop("Please check your file system permissions.")
          }
       }
    }
 })
 
+# C++ Wrappers =====================================================================================
+.rs.addFunction("convertThemeForCpp", convertThemeForCpp <- function(themePath) {
+   tryCatch(
+      withCallingHandlers(
+         .rs.convertTheme(
+            themePath,
+            add = TRUE,
+            outputLocation = NULL,
+            apply = FALSE,
+            force = TRUE,
+            globally = FALSE),
+         warning = function(w) { 
+            .Call("rs_addThemeWarning", conditionMessage(w))
+            invokeRestart("muffleWarning")
+         },
+         quiet = TRUE),
+      error = function(e) { 
+         warnings <- .Call("rs_getThemeWarnings")
```

Comment:
Oh I meant to fix that, thanks.

I know you can do that in R, but I thought it was better avoided because it can pollute the global environment? If not, it will be much cleaner that way.